WalletWasabi.Packager/Program.cs

```diff
@@ -184,12 +193,21 @@ private static async Task SignAsync()
 
 	private static async Task PublishAsync()
 	{
+		var processNamesToKill = new[] { "bitcoind", "tor" };
+		foreach (var process in processNamesToKill.SelectMany(p => Process.GetProcessesByName(p)))
+		{
+			process.Kill();
+			Console.WriteLine($"Killed '{process.ProcessName}' process.");
+		}
```

Comment:
Not too sure about this. Ideally we should kill only our `tor` & `bitcoind` instances. 

Or alternatively we should ask the user whether we can kill those processes. This seems like a reasonable compromise if we want others to use Packager safely.

Another alternative is just to say that Wasabi Packager is an internal tool and whoever works with that should know how it works. But we will most likely be unpleasantly suprised by this in the future.

Reply:
Isn't it an internal tool?

Reply:
My experience with "internal tools" is that:

* there are only few people who know how it works and can contribute
* nobody wants to touch that code after a while
* when a feature is added, it's often in an ad hoc way
* there are gotchas

So I tend to treat all code as if its importance is the same (i.e. production quality). Then more people can contribute and feel confident that it makes sense to actually try to contribute[^1].

That's why I would rather avoid treating it as an internal tool.

[^1]: Famous example of code that nobody wants to touch is X11 server.

Reply:
> My experience with "internal tools" is that:
> 
> * there are only few people who know how it works and can contribute
> * nobody wants to touch that code after a while
> * when a feature is added, it's often in an ad hoc way
> * there are gotchas
> 
> So I tend to treat all code as if its importance is the same (i.e. production quality). Then more people can contribute and feel confident that it makes sense to actually try to contribute[1](#user-content-fn-1-7c167fd217b651903a34e005426677e9).
> 
> That's why I would rather avoid treating it as an internal tool.
> 
> ## Footnotes
> 1. Famous example of code that nobody wants to touch is X11 server. [↩](#user-content-fnref-1-7c167fd217b651903a34e005426677e9)

These are all true and I agree that having internal tools is not a good practice - however for historical reasons the packager is an internal tool and not designed at all to be used by anyone. For example the constrait of having a windows and a mac hardware is unacceptable. I think the reasonable next step with this would be to move the packaging to Azure Pipelines. But that project needs devs and a completely new project - but right now we are on 2.0.

Reply:
Ah, I think I understand now. Linux version is packaged using WSL so that would be hard to do differently. 

Still my point was not "to make revolution" but rather "if we can make some parts of the code gradually more cross platform" then it's a good thing. 

* `dotnet run --onlybinaries` is a good example I guess. 
* `dotnet run` (which published fully) is hard I guess, I agree.